source/fundamentals/crud/read-operations/text.txt

```diff
@@ -0,0 +1,243 @@
+
+In MongoDB, you can perform a text search on a string field in your
+collection. 
+
+Sample Documents
+~~~~~~~~~~~~~~~~
+
+In the following examples, a student is browsing through courses to
+enroll in at their college. The documents in the ``junior_courses``
+collection represents the courses available with their intended majors: 
+
+.. code-block:: json
+
+    { "_id": 1, "name": "Forensic Science", "major": ["Forensic Science", "Criminal Justice"] }
+    { "_id": 2, "name": "Discrete Mathematical Structures for Computer Science", "major": ["Computer Science", "Mathematics"] }
+    { "_id": 3, "name": "Environmental Science", "major": ["Environmental Science", "Sustainable Management"] }
+    { "_id": 4, "name": "Image Processing in Computer Science", "major": ["Artificial Intelligence", "Computer Science"] }
+
+Text Index
+----------
+
+Create a **text index** using the ``Indexes`` builder before you perform
+a text search. A text index specifies the string or string array field
+to query during a text search. 
+
+In the following examples, the student performs text searches on the ``name``
+field in the ``junior_courses`` collection. To enable text searches on
+the ``name`` field, the student creates a text index using the following
+snippet: 
+
+.. literalinclude:: /includes/fundamentals/code-snippets/SearchText.java
+   :language: java
+   :dedent:
+   :start-after: begin textIndex
+   :end-before: end textIndex
+
+If the student wanted to index both the ``name`` and ``major`` field,
+the student can modify the previous index to a compound index. A
+compound index is a single index structure holds references to multiple
+fields within a collection's documents. To enable text searches on
+the ``name`` and ``major`` field, the student creates a compound index
+using the following snippet: 
+
+.. code-block:: java
+
+    collection.createIndex(Indexes.compoundIndex(Indexes.text("name"), Indexes.text("major")));
+
+.. note::
+
+   The student cannot create a second index because there can only be one
+   text index per collection. 
+
+For more information, see the following documentation:
+
+- :manual:`Text Indexes </core/index-text/>` Server Manual Entry
+- :ref:`Text Indexes <index-text-indexes>` in our Indexes builder guide
+- :ref:`Compound Indexes <index-compound-indexes>` in our Indexes builder guide
+
+Searching
+---------
+
+Use the ``Filters.text()`` method to perform a text search. 
+
+The ``Filters.text()`` method uses the :doc:`Filters builder
+</fundamentals/builders/filters>` to perform the text search on the
+fields indexed with the text index on the collection. The text search
+matches documents which contain one or more search terms.
+
+You can include ``TextSearchOptions`` as the second parameter to specify
+text search options such as case sensitivity. By default, there is no
+case sensitivity, meaning MongoDB matches documents regardless if the
+value is lowercase or uppercase. 
+
```

Comment:
I think the items: "certain terms" and "how to sort the relevance of your text search" are not very clear. Maybe it would be helpful just to list them by the names of the sections? Also, not sure the reader knows how to distinguish between a section and subsection. E.g.
The following sections show you how to perform the following types of text searches:
- Search Text by a Term
- Search Text by a Phrase
- Search Text with Terms Excluded
- Sort by Relevance